### DISCUSSION TITLE
Add an easy way to refer to the first cross module

### DISCUSSION THREAD

**Discussion by lefou** [com-lihaoyi/mill#2588] [Ideas]
I'm often in the situation, that I only want to run tests or tasks of one of the cross modules, typically the first. In the past, I added some proxy targets to the outer cross module, which simply calls the equally named target in the first cross instance. 

For example: To run all tests in the cross module `itest`, I can run:

```
mill itest.__.test
# or
mill itest[__].test
```

But running `test` in only the first cross value requires me to know which exact cross value exists and to call it explicitly.

```
mill itest[0.11.0-M11].test
```

**What, if we support some special syntax to be used from the CLI, to easily access the first module?**

We could adapt some systax already known from `git`.

Example: Run `test` on the first cross value of `itest`.

```
mill itest[^].test
```

We could even expand the idea to access the second or `n`th instance:

```
mill itest[^^].test
mill itest[^2].test
```

WDYT?

**Comment by lihaoyi** [comment]
Sounds like a good idea in principle. Some thoughts:

1. `.^.` is a bit cryptic. It's also a bit overloaded with the `import $file.^.bar` syntax, which means parent folder, and is itself already pretty cryptic. Maybe some other syntax, such as `cross._1` to make it an analogy to tuple syntax in Scala?

2. I feel like "first cross value" may be less good than "chosen default cross value". E.g. for cross building Scala, I often want the "default" to be neither the oldest version (barely still used/supported) nor the newest version (still unstable). Putting some arbitrary version as "first" is possible, but would mess up the sensible ordering. At work in Bazel, our cross scala libraries let people set a `default_scala_version` that will be used if no explicit version is selected. In Mill, if we chose to do that, maybe we could make `Cross` modules take an optional default (perhaps via a marker trait???), and then make `cross.foo` without square brackets be equivalent to `cross[default].foo`?

**Reply by lefou** [reply]
Yeah, a default makes sense. We already have `TaskModule.defaultCommand: String`, so we could also provide a `Cross.defaultCrossValue: T`, which could be default-implemented with the first given cross value, so nothing needs to be reconfigured, when the first cross value is already the "default", which is typically the case for me. This also makes sure a default value is not optional, so the syntax will always work on the CLI. 

I'd rather not use `itest` to select the default case `itest[0.11.0]`, as `itest` can already be used without the square brackets in some cases, e.g. when combined with wildcards. Instead we could allow the empty case `itest[]` to explicitly refer to the default.

**Reply by lefou** [reply]
* I just opened #2623
